string2.py

```diff
@@ -43,8 +51,23 @@ def not_bad(s):
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
-    """Your code goes here.  Edit this docstring."""
-    return
+    front_a = ''
+    front_b = ''
+    back_a = ''
+    back_b = ''
+    if len(a) % 2 == 0:
```

Comment:
There is a lot of 'len(a)' and 'len(b)' being done around here.  Consider refactoring so that the midpoints of each string is  computed separately, before combining their slices together